android/java/org/chromium/chrome/browser/crypto_wallet/modal/BraveWalletPanel.java

```diff
@@ -140,23 +163,100 @@ private void updateState() {
+        mBraveWalletPanelServices.getKeyringService().getSelectedAccount(CoinType.ETH, address -> {
+            String selectedAccount = "";
+            if (address != null && !address.isEmpty()) {
+                selectedAccount = address;
+                updateAccountInfo(selectedAccount);
+            } else {
+                if (mAccountsWithPermissions.size() != 0) {
+                    selectedAccount = mAccountsWithPermissions.iterator().next().address;
+                } else if (mAccountInfos.length != 0) {
+                    selectedAccount = mAccountInfos[0].address;
+                }
+                setSelectedAccount(selectedAccount);
             }
         });
     }
 
+    private void setSelectedAccount(String selectedAccount) {
+        mBraveWalletPanelServices.getKeyringService().setSelectedAccount(selectedAccount, CoinType.ETH, success -> {
+            if (success) {
+                updateAccountInfo(selectedAccount);
+            }
+        });
+    }
+
+    private void updateAccountInfo(String selectedAccount) {
+        Utils.setBlockiesBitmapResource(
+                mExecutor, mHandler, mAccountImage, selectedAccount, true);
+        for (AccountInfo accountInfo: mAccountInfos) {
+            if (accountInfo.address.equals(selectedAccount)) {
+                mAccountName.setText(accountInfo.name);
+                mAccountAddress.setText(Utils.stripAccountAddress(accountInfo.address));
+                getBalance(new AccountInfo[] {accountInfo});
+                updateAccountConnected(selectedAccount);
+                break;
+            }
+        }
+    }
+
+    private void updateAccountConnected(String selectedAccount) {
+        Iterator<AccountInfo> it = mAccountsWithPermissions.iterator();
+        boolean found = false;
+        while (it.hasNext()) {
+            if (it.next().address.equals(selectedAccount)) {
+                found = true;
+                break;
+            }
+        }
+        mBtnConnectedStatus.setText(mPopupView.getResources().getString(
+                found ? R.string.dapp_wallet_panel_connectivity_status_connected :
+                        R.string.dapp_wallet_panel_connectivity_status));
+        mBtnConnectedStatus.setCompoundDrawablesRelativeWithIntrinsicBounds(
+                found ? R.drawable.ic_check_white : 0, 0, 0, 0);
+    }
+
+    private void getBalance(AccountInfo[] selectedAccount) {
+        if (selectedAccount.length == 0) {
+            return;
+        }
+        mBraveWalletPanelServices.getJsonRpcService().getChainId(CoinType.ETH, chainId -> {
+            SingleTokenBalanceHelper singleTokenBalanceHelper =
+                    new SingleTokenBalanceHelper(
+                            mBraveWalletPanelServices.getAssetRatioService(),
+                            mBraveWalletPanelServices.getJsonRpcService());
+            singleTokenBalanceHelper.getPerAccountBalances(chainId, "",
+                    "eth", 18, selectedAccount, () -> {
```

Comment:
nit: can move "eth" and "ETH" to WalletConstants if suitable for re-usability for API calls across wallet?

Reply:
I think what we will need to do when we work on multi chain support for DApps is: some kind of a helper class that will return a native token looking on a chainId. It had to be done in all places not only here. I would leave it for now.